activemodel/lib/active_model/dirty.rb

```diff
@@ -54,7 +58,7 @@ module ActiveModel
   #
   # A newly instantiated object is unchanged:
   #
-  #   person = Person.find_by(name: 'Uncle Bob')
+  #   person = Person.new 'Uncle Bob'
```

Comment:
While this is technically correct, I don't know if this example is still meaningful after this change.

The only reason `a newly instantiated object is unchanged` is because in the constructor code doesn't use the `#name=` setter, which implements the actual tracking and has nothing to do with the API. (The Active Record implementation does the opposite, for example.)

I think my suggestion would be to move the preceding comment into the constructor, something like:

```ruby
class Person
  def initialize name = nil
    # For our purpose, we don't want newly instantiated objects to be marked as dirty,
    # so we opted not to go through the setter here:
    @name = name
  end

  ...
end

person = Person.new 'Uncle Bob'
person.changed?       # => false

...
```

Alternatively... (this seems to be closer in spirit to the original example, and matches the AR implementation):

```ruby
class Person
  def self.find_by_name(name)
    # find the person from the persistence layer and return a non-dirty object
  end

  # ...
end

person = Person.find_by(name: 'Uncle Bob')
# remove the person.changed? # => false line

...
```

Not loving either of these, so I'm open to other suggestions too.